### DISCUSSION TITLE
What is required for VenafiPS to work? Getting HTTP 401

### DISCUSSION THREAD

**Discussion by darenmaranya** [Venafi/venafips#55] [Q&A] [answered]
Hello, I work in a very large shared environment, we are attempting to connect to the TPP server but receive a HTTP 401 Unauthorized error when attempting to establish a new session. I have access to Aperture and I was told they granted Web SDK access as well, but Powershell fails to connect. Can you please tell me what is required to use New-VenafiSession?

Thanks in advance!

**Comment by gdbarron** [comment]
Hello @darenmaranya.  Are you wanting to connect via key or token based auth?  Can you provide the command you are trying to run?

**Reply by darenmaranya** [reply]
Thanks for the quick reply! I have tried a few different methods for connecting. We are in an environment which requires Smart Card authentication, so I have been trying to use the context of my admin account to run the following command:
New-VenafiSession -Server URL.of.TPPserver

**Reply by gdbarron** [reply]
Without providing credentials, authentication will occur via integrated authentication.  Has your team configured everything needeed?  https://docs.venafi.com/Docs/21.4SDK/TopNav/Content/SDK/WebSDK/t-sdk-Setup-AuthorizeAccessAD.php

**Reply by gdbarron** [reply]
Also, it is highly recommended you use token based and not key based auth.  That same section in the docs site has info on how to setup that up as well.

**Reply by gdbarron** [reply]
@darenmaranya just following up....

**Reply by gdbarron** [reply]
@darenmaranya did you still need help with this?

**Reply by darenmaranya** [reply]
I will review the documentation. I wish I could access it without an account.

**Reply by darenmaranya** [reply]
I created an account but when I access the link you posted, I get the following error:
403
App Not Assigned
Sorry, you can't access Documentation because you are not assigned this app in Okta. If you're wondering why this is happening, please [contact your administrator]

**Reply by gdbarron** [reply, marked as answer]
It would be best for you to contact support@venafi.com for issues not related to VenafiPS.
